packages/cli/src/lib/project/PluginProject.ts

```diff
@@ -124,73 +110,26 @@ export class PluginProject extends Project<PluginManifest> {
     generationSubPath?: string
   ): Promise<BindOutput> {
     const manifest = await this.getManifest();
-    const queryModule = manifest.modules.query?.module as string;
-    const queryDirectory = manifest.modules.query
-      ? this._getGenerationDirectory(queryModule, generationSubPath)
-      : undefined;
-    const mutationModule = manifest.modules.mutation?.module as string;
-    const mutationDirectory = manifest.modules.mutation
-      ? this._getGenerationDirectory(mutationModule, generationSubPath)
-      : undefined;
-
-    if (
-      queryDirectory &&
-      mutationDirectory &&
-      queryDirectory === mutationDirectory
-    ) {
-      throw Error(
-        intlMsg.lib_compiler_dup_code_folder({ directory: queryDirectory })
-      );
-    }
+    const module = manifest.module as string;
+    const moduleDirectory = this._getGenerationDirectory(
+      module,
+      generationSubPath
+    );
 
     // Clean the code generation
-    if (queryDirectory) {
-      resetDir(queryDirectory);
-    }
-
-    if (mutationDirectory) {
-      resetDir(mutationDirectory);
-    }
-
+    resetDir(moduleDirectory);
     const bindLanguage = pluginManifestLanguageToBindLanguage(
       await this.getManifestLanguage()
     );
 
     const options: BindOptions = {
       projectName: manifest.name,
-      modules: [],
+      typeInfo: composerOutput.main?.typeInfo as TypeInfo,
```

Comment:
Same here - Should we move to `module`? I think that all of these comments (the ones that are `main` -> `module`) are related. Let me know if this makes sense and I can tackle this